services/app/assets/js/widgets/components/UserInfo.jsx

```diff
@@ -0,0 +1,46 @@
+import React from 'react';
+import {OverlayTrigger, Popover} from 'react-bootstrap';
+import Loading from './Loading';
+import UserName from './UserName';
+import {loadUserStats} from '../middlewares/Users';
+
+const UserInfo = ({user, usersStats}) => {
+  let statsContent = <Loading small={true}/>;
+
+  const stats = usersStats[user.id];
+  if (stats) {
+    statsContent = (
+      <div>
+        Won: <b className="text-success">{stats.won}</b>
+        <br/>
+        Lost : <b className="text-danger">{stats.lost}</b>
+        <br/>
+        Gave up : <b className="text-warning">{stats.gave_up}</b>
+      </div>
+    );
+  }
+
+  const statsPopover = (
+    <Popover title="Stats">
+      {statsContent}
+    </Popover>
+  );
+
+  const onEnter = () => typeof stats === 'undefined' ? loadUserStats(user) : null;
+
+  return (
+    <OverlayTrigger
+      trigger={['hover', 'focus']}
+      placement="left"
+      overlay={statsPopover}
+      onEnter={onEnter}
+      shouldUpdatePosition={true}
+    >
+      <span>
+        <UserName user={user}/>
+      </span>
+    </OverlayTrigger>
+  );
+};
+
+export default UserInfo;
```

Comment:
I recommend to use [guard clauses](https://refactoring.com/catalog/replaceNestedConditionalWithGuardClauses.html) approach . And replace `statsPopover` with `renderStatsPopover` method. It would help you to make the code more declarative, i guess.